gen-ai/orchestrator-server/src/main/python/tock-llm-indexing-tools/index_documents.py

```diff
@@ -164,24 +171,31 @@ async def index_documents(args):
         index_name=index_name,
         embedding_function=embeddings
     )
-    await vector_store_factory.check_vector_store_connection()
+    # await vector_store_factory.check_vector_store_connection()
     vector_store = vector_store_factory.get_vector_store()
 
-    await embedding_and_indexing(splitted_docs, vector_store)
-
-    # Return indexing details
-    return index_name, session_uuid, len(docs), len(splitted_docs)
-
+    embedding_and_indexing(splitted_docs, vector_store)
+
+    return IndexingDetails(
+        index_name = index_name,
+        indexing_session_uuid = session_uuid,
+        documents_count = len(docs),
+        chunks_count = len(splitted_docs),
+        chunk_size = args['<chunks_size>'],
+        em_settings = em_settings,
+        vector_store_settings = vector_store_settings,
+        ignore_source = args['<ignore_source>'],
+        input_csv = args['<input_csv>'],
+        duration = datetime.now() - start_time
+    )
 
-async def embedding_and_indexing(splitted_docs: List[Document], vector_store):
+def embedding_and_indexing(splitted_docs: List[Document], vector_store):
     # Index all chunks in vector DB
     logging.debug('Index chunks in DB')
-    # Index respecting bulk_size (500 is from_documents current default: it is described for clarity only)
-    bulk_size = 500
+    bulk_size = 100  # Adjust bulk_size to suit your use case
```

Comment:
As I understand this is this only way to limit to request token rate ratio ?

Also the number of concurrent task in the asyncIO loop is an other way to limit it .. but I see that this code despite and `async main` isn't async at all ... could you remove all `async / await` code ? So that people don't misunderstand that this script isn't async and maybe add a comment to explain why.

Can you make if configurable with an optional argument using doc opt : 
```python
"""
Options:
  --embedding-bulk-size=<bs>  Number of chunks sent in each embedding requests [default: 100].
"""
```